PyInstaller/compat.py

```diff
@@ -72,9 +73,9 @@
 # disabling the compatibility mode and using python that does not properly support Big Sur still leaves find_library()
 # broken (which is a scenario that we ignore at the moment).
 # The same logic applies to macOS 12 (Monterey).
-is_macos_11_compat = _macos_ver and _macos_ver[0:2] == (10, 16)  # Big Sur or newer in compat mode
-is_macos_11_native = _macos_ver and _macos_ver[0:2] >= (11, 0)  # Big Sur or newer in native mode
-is_macos_11 = is_macos_11_compat or is_macos_11_native  # Big Sur or newer
+is_macos_11_compat = bool(_macos_ver and _macos_ver[0:2] == (10, 16))  # Big Sur or newer in compat mode
+is_macos_11_native = bool(_macos_ver and _macos_ver[0:2] >= (11, 0))  # Big Sur or newer in native mode
+is_macos_11 = bool(is_macos_11_compat or is_macos_11_native)  # Big Sur or newer
```

Comment:
Hmm, this `bool()` is reasonable, but needs to be applied only to `_macos_ver`. And then `is_macos_11` shouldn't need the explicit `bool()` anymore.